### PR TITLE
extend useRect to update for specified event

### PR DESCRIPTION
**Pull request by glie163**
useRect is a custom hook for reading the bounding rect of an element
We are using the bounding rect of a component to resize an svg within it
The rect was only updating on window resize and scroll events, we needed our component to update and resize after css transitions on the page
This PR extends useRect() so we can specify events like "transitionend" to trigger and update

### CODE REVIEW

File: packages/kerosene-ui/src/hooks/useRect.ts [hunk 1]
```diff
@@ -25,9 +25,11 @@ const DEFAULT_RECT: Rect = {
 /**
  * Custom React Hook for reading bounding rect of a DOM Element
  * @param disable When set to `true`, updating the rect will be disabled
+ * @param event Listens for specified event type and triggers update, default is "scroll"
  */
 export default function useRect(
   disable = false,
+  event = "scroll",
```
Comment (nhardy) [inline packages/kerosene-ui/src/hooks/useRect.ts]: What if we want to specify more than one event name? I think we might want this to be an array.

Comment (ojkelly) [inline packages/kerosene-ui/src/hooks/useRect.ts]: `scroll` is a very noisy event, are there any performance implications here?

It doesn't look obvious to me why we want `scroll` as the default

Comment (nhardy) [reply packages/kerosene-ui/src/hooks/useRect.ts]: `scroll` can be noisy, but along with `resize` it is one of the more common events that can change the bounding rect of an element. We're also returning the window scroll position as part of this hook as I've found that it is very commonly used together with the rect of an element.
The `update` function here is throttled with `requestAnimationFrame`, so will be called at most once per frame.
The reason for this PR was to add the capability to optionally listen for `transitionend` events, but to make this easier to use with any additional events.

File: packages/kerosene-ui/src/hooks/useRect.spec.tsx [hunk 2]
```diff
@@ -92,7 +92,7 @@ describe("useRect", () => {
 
   it("should add and remove listeners for event specified", () => {
     const Component = () => {
-      const [ref] = useRect(false, "transitionend");
+      const [ref] = useRect(false, ["transitionend"]);
```
Comment (nhardy) [inline packages/kerosene-ui/src/hooks/useRect.spec.tsx]: Do you think it's worth re-adding the check for the scroll listener for this case?

**Comment by nhardy** [comment]
Looks good to me. Not sure if I have permission to approve a PR.

**Pull request merged**